### PR TITLE
add carrier attribute to Store, Link and Line component

### LINKED ISSUE #269: Add "carrier" as an official column to lines and links
These are often used in pypsa-eur(-sec) and even required in the linopf code when constraining the transmission expansion. In `calculate_dependent_values` nans could be filled up with `n.lin{e,k}s.bus0.map(n.bus.carrier)`

### PR DESCRIPTION
**Pull request by FabianHofmann**
Closes #269 

## Changes proposed in this Pull Request
See issue

## Checklist

- [ ] ~Code changes are sufficiently documented; i.e. new functions contain docstrings and further explanations may be given in `doc`.~
- [ ] ~Unit tests for new features were added.~
- [ ] ~Newly introduced dependencies are added to `environment.yaml`, `environment_docs.yaml` and `setup.py`~
- [x] A note for the release notes `doc/release_notes.rst` of the upcoming release is included.

### CODE REVIEW

File: doc/release_notes.rst [hunk 1]
```diff
@@ -47,6 +47,8 @@ Upcoming Release
 
 * ``PyPSA`` now supports the optimization of multiple investment periods, also known as pathway optimization. With this feature, snapshots can span over multiple years or decades which are divided into investment periods. Within each investment period, assets can be added to the network. The optimization only works with ``pyomo=False``. For more information see the `example notebook <https://pypsa.readthedocs.io/en/latest/examples/multi-investment-optimisation.html>`_.       
 
+* A new attribute (column) "carrier" was added to the components ``Line``, ``Link`` and ``Store``, defining the energy carrier of the components. It's default is an empty string. When calling ``network.calculate_dependant_values``, empty carriers are replaced by the carriers of the buses to which the components are attached.
```
Comment (lisazeyen) [inline doc/release_notes.rst]: add storage unit

**Review by lisazeyen — APPROVED** [review]

**Comment by FabianHofmann** [comment]
Hey @lisazeyen, carrier was already an attribute for `StorageUnit`, there is now a duplicated row in component_attrs/storage_units.csv. Either we revert the changes or force delete the commits

**Comment by lisazeyen** [comment]
Sorry @FabianHofmann, I have missed that it was already an attribute!

**Comment by FabianHofmann** [comment]
No problem :)

**Pull request merged**